### PR TITLE
Add translate/review story button to card

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link
[Add "translate/review book" button to story card](https://www.notion.so/uwblueprintexecs/Add-translate-review-book-button-to-story-card-4f3735126cdf46c9a36d0a900af6ca6c)

## Implementation description
* Added translate/review button to story card
* Created primary/secondary button components
* Pressing button makes corresponding calls to createStoryTranslation / assignUserAsReviewer
* Note: currently hardcoding values that i need passed into StoryCard component in StoryList; these will need to actually be passed in at some point

## Steps to test
1. Seed database with users and stories/story_translations; log into the frontend and navigate to `/stories`
```
$ INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('First', 'User', <firebase_auth_id>, 'User', '{"HINDI":3}');
$ INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Second', 'User', <firebase_auth_id>, 'User', '{"HINDI":3}');
$ INSERT INTO stories (title, description, youtube_link, level) VALUES ('title', 'description', 'link', 1);
$ INSERT INTO story_translations (story_id, language, stage, translator_id, reviewer_id) VALUES (<story_id>, 'HINDI', 'TRANSLATE', <first_user_id>, null);
```
*** NOTE: values for storyId, storyTranslationId and language are hardcoded in the StoryList component--these will probably need to be changed to match the test data you insert

_test review:_
image
2. Verify that the review button is only displayed when storyTranslationId passed in
3. Verify that clicking the button redirects to `/review/{story translation id}`
4. Verify the changes in the database (currently logged in user should be assigned as the reviewer for that story in `story_translations`)
```
$ SELECT * FROM story_translations;
```

_test translate:_
image
5. Verify that translate button is displayed when storyTranslationId is not passed in
6. Verify that clicking the button redirects to `translation/{story id}`
7. Verify the changes in the database (a new row should be added to `story_translations` with the corresponding story info + the currently logged in user id as the translator_id)
```
$ SELECT * FROM story_translations;
```

## What should reviewers focus on?
* StoryCard, PrimaryButton, SecondaryButton components
* Should the gql calls in StoryCard be separated out into another file?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/story/StoryCard.tsx [hunk 1]
```diff
@@ -29,9 +40,85 @@ const StoryCard = ({
 
     return originalYoutubeLink.replace("watch?v=", "embed/");
   };
+  const handleError = (errorMessage: string) => {
+    alert(errorMessage);
+  };
+
+  const ASSIGN_REVIEWER = gql`
```
Comment (gaoxk) [inline frontend/src/components/story/StoryCard.tsx:47]: nit: move `ASSIGN_REVIEWER, AssignReviewer, CREATE TRANSLATION, CreateTranslation` out of `StoryCard `(just before `StoryCardProps `is good). I'll make it a todo to centralize gql calls in a different file in the future.

**Review by gaoxk — CHANGES_REQUESTED** [review]
looks great so far!

File: frontend/src/components/story/StoryCard.css [hunk 2]
```diff
@@ -5,7 +5,7 @@
     flex-direction: row;
     flex-wrap: wrap;
     margin: 10px 0px 10px 10px;
-    min-width: 800px;
+    min-width: 924px;
```
[the remark this answers is not in the document]
Comment (jennchenn) [reply frontend/src/components/story/StoryCard.css:8]: i was basing it off the pixel sizing in figma--with 800px the buttons didn't seem to fit properly into the card at the width given but happy to change it!!->might need some guidance on how big/small to make things though

[the remark this answers is not in the document]
Comment (gaoxk) [reply frontend/src/components/story/StoryCard.css:8]: lets go with figma :)

**Review by gaoxk — APPROVED** [review]
🚢

**Pull request merged**